circuit/environment/src/helpers/r1cs.rs

```diff
@@ -91,9 +93,33 @@ impl<F: PrimeField> R1CS<F> {
         self.counter.add_constraint(constraint);
     }
 
-    /// Returns `true` if all constraints in the environment are satisfied.
-    pub fn is_satisfied(&self) -> bool {
-        self.constraints.iter().all(|constraint| constraint.is_satisfied())
+    /// Returns `true` if both
+    /// - all constraints in the environment are satisfied.
+    /// - all constraints use variables corresponding to the declared variables
+    pub fn is_satisfied(&self) -> Result<bool> {
```

Comment:
We should return `bool` and use `.all()` with a `.map_or_else` or equiv.

Reply:
🙏 Fixed, squashed and pushed.